src/cylinder.jl

```diff
@@ -43,8 +43,9 @@ function endcircles(s::Cylinder{3})
     b2 = cross(b1, s.a)
     axes = [s.a[1] b1[1] b2[1]; s.a[2] b1[2] b2[2]; s.a[3] b1[3] b2[3]]
     d = 2*s.r
-    return(Ellipsoid(s.c + s.a*s.h2, SVector(0.0, d, d), axes),
-           Ellipsoid(s.c - s.a*s.h2, SVector(0.0, d, d), axes))
+    et = Ellipsoid(s.c + s.a*s.h2, SVector(0.0, d, d), axes)::Ellipsoid{3,Void,9}  # top disk
+    eb = Ellipsoid(s.c - s.a*s.h2, SVector(0.0, d, d), axes)::Ellipsoid{3,Void,9}  # bottom disk
```

Comment:
The outer constructors will definitely be type-unstable if passed a `Vector` like `[0,0]`, since the length of that vector (hence the dimensionality) can only be determined at runtime.  This is okay .... what matters is the type-stability in critical inner loops, not whether runtime dispatch occurs at some outer level.

Adding type-stable constructors for `SVector` etcetera seems like the way to go if we need this.  It's a shame that we can't use a single constructor for both cases, but this should be fine for now.